kiva/tests/drawing_tester.py

```diff
@@ -186,17 +186,23 @@ def assertImageSavedWithContent(self, filename):
         """ Load the image and check that there is some content in it.
         """
         image = numpy.array(Image.open(filename))
-        # default is expected to be a totally white image
+        # Default is expected to be a totally white image.
+        # Therefore we check if the whole image is not white.
+
+        # Previously this method checked for red pixels. However this is
+        # not [currently] possible with the quartz backend because it writes
+        # out image with premultiplied alpha and none of its pixels are the
```

Comment:
This is from the `if 'A' in mode:` block in the above file correct?

Reply:
It is not. The premultiplied alpha setting happens in the `__init__` method of the graphics context. I couldn't find a way to have normal alpha, so I changed the test and added this note.

Reply:
So there are a few constraints here:
* `alpha_info` can only be `kCGImageAlphaPremultipliedLast` or `kCGImageAlphaPremultipliedFirst` if `save()` is called on a graphics context.
* From the "Supported Pixel Formats" table on [this page](https://developer.apple.com/library/archive/documentation/GraphicsImaging/Conceptual/drawingwithquartz2d/dq_context/dq_context.html#//apple_ref/doc/uid/TP30001066-CH203), you can see that only 8, 16, and 32bit pixel formats are supported, so we can't even make a packed 24bit pixel buffer which would make alpha-free saving possible.

We can definitely implement saving RGB/BGR images, but it's a question of time allocation.

Reply:
I see, thank you!